docs/core_docs/docs/integrations/vectorstores/memory.ipynb

```diff
@@ -0,0 +1,382 @@
+{
+ "cells": [
+  {
+   "cell_type": "raw",
+   "id": "1957f5cb",
+   "metadata": {
+    "vscode": {
+     "languageId": "raw"
+    }
+   },
+   "source": [
+    "---\n",
+    "sidebar_label: In Memory\n",
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ef1f0986",
+   "metadata": {},
+   "source": [
+    "# MemoryVectorStore\n",
+    "\n",
+    "MemoryVectorStore is an in-memory, ephemeral vectorstore that stores embeddings in-memory and does an exact, linear search for the most similar embeddings. The default similarity metric is cosine similarity, but can be changed to any of the similarity metrics supported by [ml-distance](https://mljs.github.io/distance/modules/similarity.html).\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "36fdc060",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "To access the in memory vector store you'll need to install the `langchain` integration package.\n",
+    "\n",
+    "```{=mdx}\n",
+    "import IntegrationInstallTooltip from \"@mdx_components/integration_install_tooltip.mdx\";\n",
+    "import Npm2Yarn from \"@theme/Npm2Yarn\";\n",
+    "\n",
+    "<IntegrationInstallTooltip></IntegrationInstallTooltip>\n",
+    "\n",
+    "<Npm2Yarn>\n",
+    "  langchain\n",
+    "</Npm2Yarn>\n",
+    "```\n",
+    "\n",
+    "### Credentials\n",
+    "\n",
+    "If you want to get automated tracing of your model calls you can also set your [LangSmith](https://docs.smith.langchain.com/) API key by uncommenting below:\n",
+    "\n",
+    "```typescript\n",
+    "// process.env.LANGCHAIN_TRACING_V2=\"true\"\n",
+    "// process.env.LANGCHAIN_API_KEY=\"your-api-key\"\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "93df377e",
+   "metadata": {},
+   "source": [
+    "## Instantiation\n",
+    "\n",
+    "In this example we'll use OpenAIEmbeddings, however you can use any embeddings you like."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "dc37144c-208d-4ab3-9f3a-0407a69fe052",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import { MemoryVectorStore } from 'langchain/vectorstores/memory'\n",
+    "import { OpenAIEmbeddings } from '@langchain/openai'\n",
+    "\n",
+    "const vectorStore = new MemoryVectorStore(new OpenAIEmbeddings())"
```

Comment:
I would show how to initialize `.fromDocuments` as well as from an existing store